src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/EasyCreditMapper.php

```diff
@@ -90,7 +93,47 @@ protected function getDataSubArray(ComputopEasyCreditPaymentTransfer $cardPaymen
         $dataSubArray[ComputopApiConfig::ETI_ID] = $this->config->getEtiId();
         $dataSubArray[ComputopApiConfig::IP_ADDRESS] = $cardPaymentTransfer->getClientIp();
         $dataSubArray[ComputopApiConfig::SHIPPING_ZIP] = $cardPaymentTransfer->getShippingZip();
+        $dataSubArray[ComputopApiConfig::SHIPPING_CITY] = $cardPaymentTransfer->getShippingCity();
+        $dataSubArray[ComputopApiConfig::SHIPPING_COUNTRY_CODE] = $cardPaymentTransfer->getShippingCountryCode();
+        $dataSubArray[ComputopApiConfig::SHIPPING_STREET] = $cardPaymentTransfer->getShippingStreet();
+        $dataSubArray[ComputopApiConfig::SHIPPING_STREET_NUMBER] = $cardPaymentTransfer->getShippingStreetNumber();
 
         return $dataSubArray;
     }
+
+    /**
+     * @param array $addressData
+     * @param \Generated\Shared\Transfer\ComputopEasyCreditPaymentTransfer $computopEasyCreditPaymentTransfer
+     *
+     * @return \Generated\Shared\Transfer\ComputopEasyCreditPaymentTransfer
+     */
+    protected function mapAddressDataToEasyCreditPayment(
+        array $addressData,
+        ComputopEasyCreditPaymentTransfer $computopEasyCreditPaymentTransfer
+    ): ComputopEasyCreditPaymentTransfer {
+
+        $computopEasyCreditPaymentTransfer->setShippingCity($addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY]);
```

Comment:
mb smth like `$addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY] ?? null`?

Reply:
```
    protected function getAdditionalAddressData(QuoteTransfer $quoteTransfer): array
    {
        $addressTransfer = $quoteTransfer->getBillingSameAsShipping() ? $quoteTransfer->getBillingAddress() : $quoteTransfer->getShippingAddress();

        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_CITY] = $addressTransfer->getCity();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_STREET] = $addressTransfer->getAddress1();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_STREET_NUMBER] = $addressTransfer->getAddress2();
        $addressData[ComputopEasyCreditPaymentTransfer::SHIPPING_COUNTRY_CODE] = $addressTransfer->getIso2Code();
}
```
```
$addressData = $this->getAdditionalAddressData($quoteTransfer);
$computopPaymentTransfer = $this->mapAddressDataToEasyCreditPayment($addressData, $computopPaymentTransfer);
```

So this keys will always exist. I think it`s not needed to use ternar operation in this case. What do you think ?

Reply:
What about using `$transfer->fromArray($array);` ?

Reply:
There is already related property in `$computopEasyCreditPaymentTransfer` e.g. `shippingZip`.
So i`m following same structure for other adderrs data, `shippingCity` , `shippingStreet` ..

`$transfer->fromArray($array);` will not work this way due to different proprites naming: `city`, `address1`.

cc: @aleksey-kotsuba